app/javascript/src/BackendApis/components/NewBackendModal.jsx

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+import * as React from 'react'
+
+import {
+  BaseSizes,
+  Modal,
+  Spinner,
+  Title,
+  TitleLevel
+} from '@patternfly/react-core'
+// $FlowFixMe[missing-export] To fix this: rename src/Form si that name-mapper does not mess up
+import { NewBackendForm } from 'BackendApis'
+
+import './NewBackendModal.scss'
+
+import type { Backend } from 'Types'
+
+type Props = {
+  backendsPath: string,
+  isOpen?: boolean,
+  onClose: () => void,
+  onCreateBackend: (Backend) => void
+}
+
+const NewBackendModal = ({ backendsPath, isOpen = false, onClose, onCreateBackend }: Props): React.Node => {
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [errors, setErrors] = React.useState()
+
+  const handleOnAjaxComplete = (_event, xhr: { responseText: string }, status: string) => {
+    setIsLoading(false)
+
+    if (status === 'success') {
+      const backend = JSON.parse(xhr.responseText)
+      onCreateBackend(backend)
+    } else if (status === 'error') {
+      const errors = JSON.parse(xhr.responseText)
+      setErrors(errors)
+    }
+  }
+
+  React.useEffect(() => {
+    $('form#new_backend_api_config')
```

Comment:
Why we need jQuery here, can we add listeners in other way?
If complicated and out of scope, we could add a todo

Reply:
This has to do with the way Rails makes async calls. I think we haven't done this with React before but the library is injected into jQuery so I'm afraid this is how it's done.. ugly yes. We can't even import $ from node_modules because that's a different version and it won't include jquery-ujs.

In Rails 5 there's a new module that replaces jquery-ujs, however porta uses it extensively and migrating to rails-ujs (the new module) would be a massive task. There are already some todos pointing to this. But again, I think it's not realistic to think we're going to "fix" this anytime soon. At least we'd need to migrate everything to React and stop using colorbox and the rest of old libraries.

Reply:
Actually I just saw there's a `$FlowFixMe` in the next line. These are to be treated as TODO.